File: level1/ex01.js
function solution(lottos, win_nums) {
  const answer = [];

  let lottos_miss = 0;

  // 0 개수 세기
  lottos.forEach(value => {
    if (value === 0) lottos_miss++
  });

  let lottos_hits = 0;

  // 0이 아닌 번호 중 맞은 개수 세기
  lottos.forEach(value => {
    if (value !== 0 && win_nums.includes(value)) lottos_hits++;
  })

  function rank(hits) {
    if (hits <= 1) return 6;
    else return 6 - hits + 1;
  }

  answer.push(rank(lottos_hits + lottos_miss));
  answer.push(rank(lottos_hits));

  return answer;
}

File: level1/ex01.py
def solution(lottos, win_nums):
    # 알아볼 수 없는 번호 개수
    unknown = lottos.count(0)
    # 일치하는 번호 개수
    matched = len(set(lottos) & set(win_nums))
    # 최고 등수
    high_rank = min(6, 7 - matched - unknown)
    # 최저 등수
    low_rank = min(6, 7 - matched)

    return [high_rank, low_rank]